gadget/gadget.go

```diff
@@ -189,11 +189,16 @@ type VolumeContent struct {
 	Unpack bool `yaml:"unpack"`
 }
 
+func (vc VolumeContent) ResolvedSource() string {
+	// TODO: implement resolved sources
+	return vc.UnresolvedSource
+}
+
 func (vc VolumeContent) String() string {
 	if vc.Image != "" {
 		return fmt.Sprintf("image:%s", vc.Image)
 	}
-	return fmt.Sprintf("source:%s", vc.Source)
+	return fmt.Sprintf("source:%s", vc.UnresolvedSource)
```

Comment:
should this be `vc.ResovledSource()` ?

Reply:
I was wondering about this one too, I made it UnresolvedSource because that is shorter and more univeral, i.e. it would be `$kernel:pi-dtbs/overlays` right now. With "ResolvedSource()" it would be `/snap/pi-kernel/1234/overlays` instead. Maybe we will need both eventually when we log this in debug/info messages.

Reply:
if we have a non costly way of telling whether the source needs resolving (maybe `!strings.HasPrefix(s, "$")` is enough?), we could do:
```
if !needsResolving {
    return fmt.Sprintf("source:%s", vc.UnresolvedSource)
}
res := vc.ResolvedSource()
return fmt.Sprintf("source:%s (%s)", vc.UnresolvedSource, res)
```

Reply:
I think using just UnresolvedSource for now is OK

Reply:
I like @bboozzoo's recommendation eventually, but not needed for this PR I think